### PR TITLE
Move QgsProviderSublayersDialog to GUI

### PR DESCRIPTION
**Pull request by manisandro**
Make it accessible to third-party code.

### CODE REVIEW

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -1489,6 +1491,9 @@ set(QGIS_GUI_UI_HDRS
   ${CMAKE_CURRENT_BINARY_DIR}/../ui/ui_qgstemporalcontrollerwidgetbase.h
   ${CMAKE_CURRENT_BINARY_DIR}/../ui/ui_qgsprocessingtinmeshdatawidgetbase.h
   ${CMAKE_CURRENT_BINARY_DIR}/../ui/ui_qgsexpressionpreviewbase.h
+  ${CMAKE_CURRENT_BINARY_DIR}/../ui/ui_qgsprovidersublayersdialogbase.h
+  ${CMAKE_CURRENT_BINARY_DIR}/../ui/ui_qgsdatumtransformdialogbase.h
+  ${CMAKE_CURRENT_BINARY_DIR}/../ui/ui_qgscoordinateoperationwidgetbase.h
```
Comment (m-kuhn) [inline src/gui/CMakeLists.txt:1496]: Unrelated?

File: src/gui/qgsprovidersublayersdialog.h [hunk 2]
```diff
@@ -48,24 +62,48 @@ class QgsProviderSublayerDialogModel : public QgsProviderSublayerModel
...
+     */
     QgsProviderSublayersDialog( const QString &uri,
                                 const QString &filePath,
                                 const QList< QgsProviderSublayerDetails> initialDetails = QList< QgsProviderSublayerDetails>(),
                                 const QList< QgsMapLayerType > &acceptableTypes = QList< QgsMapLayerType >(),
                                 QWidget *parent SIP_TRANSFERTHIS = nullptr,
                                 Qt::WindowFlags fl = Qt::WindowFlags() );
 
+    /*
+     * Set list of non-layer items.
```
Comment (m-kuhn) [inline src/gui/qgsprovidersublayersdialog.h:88]: If you happen to know, what this refers to, a more verbose version would be appreciated

File: src/gui/qgsprovidersublayersdialog.h [hunk 3]
```diff
@@ -28,17 +29,30 @@ class QgsProviderSublayerModel;
...
+    /**
+     * Constructor.
+     */
     QgsProviderSublayerDialogModel( QObject *parent = nullptr );
 
     QVariant data( const QModelIndex &index, int role ) const override;
     Qt::ItemFlags flags( const QModelIndex &index ) const override;
 
+    /**
+     * Sets whether geometry types are resolved.
```
Comment (m-kuhn) [inline src/gui/qgsprovidersublayersdialog.h:54]: If you happen to know, what this refers to, a more verbose version would be appreciated

**Comment by m-kuhn** [comment]
Looks good to me

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (manisandro) [reply src/gui/CMakeLists.txt:1496]: Semi-related - included headers refer to these headers. Though might indeed also come from another include path from qgsapplayerhandling.cpp which I was porting to KADAS. I can split it out to a separate commit if preferred.

File: src/gui/qgsprovidersublayersdialog.h (hunk 2 above)
Comment (manisandro) [reply src/gui/qgsprovidersublayersdialog.h:88]: Not directly familiar with the code, but I can try and formulate a better version. But I'm currently at loss why the documentation test still believes that these methods are undocumented?

File: src/gui/CMakeLists.txt (hunk 1 above)
Comment (m-kuhn) [reply src/gui/CMakeLists.txt:1496]: No need to, just wanted to be sure it's not an accident.

File: src/gui/qgsprovidersublayersdialog.h (hunk 2 above)
Comment (nyalldawson) [reply src/gui/qgsprovidersublayersdialog.h:88]: For the record, the model docs explain further what non layer items are. Currently only embedded projects are using these.

**Comment by manisandro** [comment]
I've addressed the remaining issues, MinGW dependency issues should resolve itself shortly.

**Review by m-kuhn — APPROVED** [review]
LGTM, thanks
